### ISSUE TITLE
NoSuchMethodError: The getter 'isInSystemLibrary' was called on null.

### ISSUE THREAD

**Issue by ravimittal16** [Milad-Akarie/auto_route_library#22]
I am getting a build error (below), while trying to add a new route.

> [SEVERE] auto_route_generator:autoRouteGenerator on lib/routes/router.dart:
> Error running AutoRouteGenerator
> NoSuchMethodError: The getter 'isInSystemLibrary' was called on null.
> Receiver: null
> Tried calling: isInSystemLibrary

```
@autoRouter
class $Router {
  @initial
  LoginPage loginPage;
  DashboardPage dashboardPage;
  DashboardDetailPage dashboardDetailPage;
}
```

```
class DashboardDetailPage extends StatelessWidget {
  // final dynamic itemPayload;
  final String selectedType;

  const DashboardDetailPage({Key key, this.selectedType}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return Container(child: Text("HELL"));
  }
}
```

if I comment **DashboardDetailPage**, then everything works fine. I want to add a dynamic route argument in this class/page.

**Comment by iscriptology** [comment]
Experiencing the same issue here

**Comment by Milad-Akarie** [comment]
@iscriptology  @ravimittal16  Are you passing in a call back function to your route widgets?

**Comment by ravimittal16** [comment]
No, Although I have fixed the issue but this was happening due to having a dynamic object final `dynamic itemPayload;` (parameter) dependency on the widget (route). When I updated the dynamic object to a strongly typed object, everything build successful.

**Comment by Milad-Akarie** [comment]
fixed in v0.3.0 dynamic params will not cause the generator to crash although it's good practice to type hint all of your params. 

Thanks for your efforts.
